utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -49,6 +49,14 @@ Status WritePreparedTxnDB::Initialize(
   SequenceNumber prev_max = max_evicted_seq_;
   SequenceNumber last_seq = db_impl_->GetLatestSequenceNumber();
   AdvanceMaxEvictedSeq(prev_max, last_seq);
+  // Create a gap between max and the next snapshot. This simplifies the logic
+  // in IsInSnapshot by not having to consider the special case of max ==
+  // snapshot after recovery. This is tested in IsInSnapshotEmptyMapTest.
```

Comment:
What will be the problem if `max == snapshot` after recovery? Just asking.

Reply:
The IsInSnapshot algorithm has a check:
```
if (max_evicted_seq < snapshot_seq) {
// then check old_commit_map_
```
which is valid when max advances snapshot_seq. However during recovery max_evicted_seq == snapshot_seq and yet the snapshot is not added to old_commit_map.
To avoid adding more if-then-elses here to account for this case too, we simply make sure that max_evicted_seq < snapshot_seq right after recovery.